src/Servers/Kestrel/Core/src/Internal/Http/HttpRequestHeaders.cs

```diff
@@ -59,11 +60,14 @@ protected override void ClearFast()
         else
         {
             // If we are reusing headers, store the currently set headers for comparison later
-            _previousBits = _bits;
+            // Psuedo header bits were cleared at the start of a request to hide from the user.
+            // Keep those values for reuse.
+            _previousBits = _bits | _psuedoBits;
```

Comment:
Not including some the pseudo headers in `_bits` means that their values don't get cleared when `!ReuseHeaderValues`. It looks like we currently have a similar issue with `ClearInvalidH2H3Headers()`, but at least those are response headers that should never be proxied, so that's only a problem for buggy apps.

Reply:
Does this fix what you mentioned?

```diff
if (!ReuseHeaderValues)
{
    // If we aren't reusing headers clear them all
-   Clear(_bits);
+   Clear(_bits | _psudeoBits);
}
```